Documentation/tutorial/intro markdown lite.md

```diff
@@ -0,0 +1,183 @@
+# Markdown Lite
+
+## Introduction
+
+Markdown lite is a simple markdown tool to markup `md` file.
+
+## Design goal
+
+We write this tool for good extensiblity, so our implements should obay following order:
+
+1.  Extensiblity:
+    * Support markdown syntax extension.
+    * Support validation extension.
+2.  Correctness:
+    We follow GFM syntax, but when some rule is too hard to implement, just breaking.
+3.  Performance:
+    We almost don't care performance.
+
+## Steps
+
+There are three step when calling [markup method](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownEngine.Markup(System.String,System.String)):
+* [Parse](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownParser)
+* [Rewrite](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownTokenRewriter) or [validate](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownTokenValidator)
+* [Render](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownRenderer)
+
+### Parse
+
+In this step, it will parse markdown text to [models](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownToken).
+And in this phase, it is [rule](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownRule) based.
+Then a set of rule is called [context](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownContext).
+
+For example,
+[heading token](xref:Microsoft.DocAsCode.MarkdownLite.MarkdownHeadingBlockToken) is created by [heading rule](xref:Microsoft.DocAsCode.MarkdownLite.MarkdownHeadingBlockRule),
+the heading rule is belong to [block context](xref:Microsoft.DocAsCode.MarkdownLite.MarkdownBlockContext).
+
+### Rewrite or validate
+
+In this step, it will walk through all [models](xref:Microsoft.DocAsCode.MarkdownLite.IMarkdownToken),
+we can changed it to another, or just validate the content.
+
+For example, we can create a rewriter to change all heading token with depth + 1:
+
+```csharp
+MarkdownTokenRewriterFactory.FromLambda<IMarkdownRewriteEngine, MarkdownHeadingBlockToken>(
+    (engine, token) => new MarkdownHeadingBlockToken(token.Rule, token.Context, token.Content, token.Id, token.Depth + 1, token.SourceInfo);
+```
+
+### Render
+
+In this step, it will render to text content (html format for normal).
+To simplify extension, we create an [adapter](xref:Microsoft.DocAsCode.MarkdownLite.MarkdownRendererAdapter),
+any class with following method will treat as render method:
```

Comment:
What about: Any class with the following method will be treated as renderer class, and the method will be treated as render method.